activerecord/lib/active_record/attribute_methods/write.rb

```diff
@@ -17,7 +17,7 @@ def define_method_attribute=(name)
             generated_attribute_methods.module_eval <<-STR, __FILE__, __LINE__ + 1
               def __temp__#{safe_name}=(value)
                 name = ::ActiveRecord::AttributeMethods::AttrNames::ATTR_#{safe_name}
```

Comment:
They seem to be equivalent, or at least close enough:

### Before:

```
Warming up --------------------------------------
           attribute   134.983k i/100ms
          attribute=    35.139k i/100ms
Calculating -------------------------------------
           attribute      2.338M (± 3.3%) i/s -     11.744M in   5.029545s
          attribute=    401.747k (± 3.4%) i/s -      2.038M in   5.079005s
```

### After:

```
Warming up --------------------------------------
           attribute   132.856k i/100ms
          attribute=    34.014k i/100ms
Calculating -------------------------------------
           attribute      2.337M (± 3.3%) i/s -     11.691M in   5.008654s
          attribute=    399.083k (± 2.8%) i/s -      2.007M in   5.032575s
```

Without `freeze`, for comparison:

```
Warming up --------------------------------------
           attribute   133.407k i/100ms
          attribute=    33.575k i/100ms
Calculating -------------------------------------
           attribute      2.201M (± 2.9%) i/s -     11.073M in   5.034287s
          attribute=    388.022k (± 2.5%) i/s -      1.947M in   5.021768s
```

It _might_ be worth switching to the form you suggested, as it's much simpler to read and avoids creating a bunch of extra constants. I'd see this as a different patch, but if you feel strongly about it I'll wedge it in here 😄